roadmap/implementers-guide/src/glossary.md

```diff
@@ -2,44 +2,71 @@
+- **Preimage:** In our context, if `H(X) = Y` where `H` is a hash function and `Y` is the hash, then `X` is the hash preimage.
+- **Proof-of-Validity (PoV):** A stateless-client proof that a parachain candidate is valid, with respect to some validation function.
+- **PVF:** Parachain Validation Function. The validation code that is run by validators on parachains or parathreads. See the [PVF glossary][1].
+- **Relay Parent:** A block in the relay chain, referred to in a context where work is being done in the context of the state at this block.
+- **Router:** The router module is a meta module that consists of three runtime modules responsible for routing messages between paras and the relay chain. The three separate runtime modules are: Dmp, Ump, Hrmp, each responsible for the respective part of message routing.
+- **Runtime:** The relay-chain state machine.
+- **Runtime Module:** See Module.
+- **Runtime API:** A means for the node-side behavior to access structured information based on the state of a fork of the blockchain.
+- **Secondary Checker:** A validator who has been randomly selected to perform secondary approval checks on a parablock which is pending approval.
+- **Subsystem:** A long-running task which is responsible for carrying out a particular category of work.
+- **UMP:** (Upward Message Passing) A vertical message passing mechanism from a parachain to the relay chain.
+- **Validator:** Specially-selected node in the network who is responsible for validating parachain blocks and issuing attestations about their validity.
+- **Validation Function:** A piece of Wasm code that describes the state-transition function of a parachain.
+- **VMP:** (Vertical Message Passing) A family of mechanisms that are responsible for message exchange between the relay chain and parachains.
+- **XCMP:** (Cross-Chain Message Passing) A type of horizontal message passing (i.e. between parachains) that allows secure message passing directly between parachains and has minimal resource requirements from the relay chain, thus highly scalable.
+
+## PVF
+
+The PVF functionality involves several processes which may be potentially
+confusing:
+
+- **Prechecking:** This is the process of initially checking the PVF when it is
+  first added. We attempt *preparation* of the PVF and make sure it succeeds
+  within a given timeout.
+- **Execution:** This actually executes the PVF. The node may not have the
+  artifact from prechecking, in which case this process also includes a
+  *preparation* job. The timeout for preparation here is more lenient than when
+  prechecking.
+- **Preparation:** This is the process of preparing the WASM blob and includes
+  both *prevalidation* and *compilation*. As prevalidation is pretty minimal
+  right now, preparation mostly consists of compilation. Note that *prechecking*
+  just consists of preparation, whereas *execution* will also prepare the PVF if
+  the artifact is not already found.
```

Comment:
I don't get this last sentence. prechecking == preparation && execution == preparation?

Reply:
Yeah that is confusing, I will try to reword it a bit. How about this: 

Prechecking *only* tries to prepare the artifact, with a strict timeout. Execution will also prepare (if the artifact is not found), but with a more lenient timeout, *in addition* to then executing the compiled code.

Reply:
Ended up removing this sentence.